pkg/controller/virtualmachine/virtualmachine_controller.go

```diff
@@ -86,15 +84,79 @@ func (r *ReconcilePolicy) Reconcile(request reconcile.Request) (reconcile.Result
 	err := r.Get(context.TODO(), request.NamespacedName, instance)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			err := r.poolManager.ReleaseVirtualMachineMac(fmt.Sprintf("%s/%s", request.Namespace, request.Name))
-			if err != nil {
-				logging.Printf(logging.ErrorLevel, "failed to release mac for virtual machine %s: %v", request.NamespacedName, err)
-			}
 			return reconcile.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
 
-	return reconcile.Result{}, nil
+	if instance.ObjectMeta.DeletionTimestamp.IsZero() {
+		log.V(1).Info("The object is not being deleted")
```

Comment:
"The object is not being deleted"
Could we have a hint of the reason?
Maybe something like:
"The object was not deleted. Finalizers were not run correctly."

Reply:
Hi @mmirecki . Thank you for the notes. what do you mean by "reason"? 
the fact that the object is not being deleted  is not an error - if the object is not being deleted then we either add a finalizer, or the object already have a finalzier so we leave it as it is.
can you clarify on this? 
 the purpose of this log is just to show "where we stand" in the reconcile function.

Reply:
makes sense, my wrong here